lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+require_relative "./throttled_retriever"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        def self.prepended(base)
+          base.prepend(ThrottledRetriever)
+        end
+
+        private
+
+        # Calls SuperFetch UnitOfWork's requeue to remove the job from the
+        # temporary queue and push job back to the head of the queue, so that
+        # the job won't be tried immediately after it was requeued (in most cases).
+        #
+        # @note This is triggered when job is throttled.
+        #
+        # @return [void]
+        def requeue_throttled(work)
+          # SuperFetch UnitOfWork's requeue will remove it from the temporary
+          # queue and then requeue it, so no acknowledgement call is needed.
+          work.requeue
+        end
+
+        # Returns list of non-paused queues to try to fetch jobs from.
+        #
+        # @note It may return an empty array.
+        # @return [Array<Array(String, String)>]
+        def active_queues
+          # Create a hash of throttled queues for fast lookup
+          throttled_queues = Throttled.cooldown&.queues&.to_a&.to_h { [_1, true] }
+          return super unless throttled_queues
```

Comment:
I think we can check if throttled queues is nil or empty here:
```suggestion
          return super if throttled_queues.nil? || throttled_queues.empty?
```